internal/decomposer/decomposer.go

```diff
@@ -129,59 +129,12 @@ func (decomposer *Decomposer) ExecuteStep(step cacao.Step, scopeVariables cacao.
 	variables.Merge(step.StepVariables)
 
 	switch step.Type {
-	case "action":
-		return decomposer.ExecuteActionStep(step, variables)
+	case cacao.StepTypeAction:
+		details := action.StepDetails{Step: step, Targets: decomposer.playbook.TargetDefinitions, Auth: decomposer.playbook.AuthenticationInfoDefinitions, Agent: decomposer.playbook.AgentDefinitions[step.Agent], Variables: variables}
```

Comment:
I personally find slightly confusing the naming of the struct fields here, because some are step-specific and some are note.

In StepDetails, we are inserting *all* targets definitions and *all* authinfo. I would then name the struct fields something more like PlaybookTargets, PlaybookAuthInfoDefinitions, and StepAgent.

Reply:
maybe playbook metadata is better?

Reply:
Not sure what you mean. You mean instead of StepDetails? The role of the "details" variable is still to provide step details and related step context right? And also there's already the "metadata" var being used.